visualization/src/resource-visualization.cc

```diff
@@ -419,4 +432,160 @@ void visualizeReprojectedDepthResourcePerRobot(
+        line_segment.from = cur_pos;
+        line_segment.to = cur_pos;
+        line_segment.color = color;
+        line_segment.scale = 0.4;
+        line_segment.alpha = 1.0;
+
+        Pose pose;
+        pose.G_p_B = T_G_S.getPosition();
+        pose.G_q_B = T_G_S.getRotation().vector();
+        pose.id = 0;
+        pose.scale = 1.8;
+        pose.line_width = 0.5;
+        pose.alpha = 1.0;
+
+        if (point_cloud_counter > 1) {
+          line_segment.from = (*resources)[previous_ts_ns].edge.to;
+        }
+        SequentialPointCloud& cur = (*resources)[ts_ns];
+        cur.point_cloud = std::move(points_G);
+        cur.edge = std::move(line_segment);
+        cur.pose = std::move(pose);
+        cur.mission = mission_counter;
+
+        previous_ts_ns = ts_ns;
+        return;
+      };
+
+  vi_map::MissionIdList mission_ids;
+  mission_ids.emplace_back(mission_id);
+  depth_integration::integrateAllDepthResourcesOfType(
+      mission_ids, input_resource_type,
+      FLAGS_vis_pointcloud_reproject_depth_maps_with_undistorted_camera, vi_map,
+      integration_function);
+}
+
+void visualizeReprojectedDepthResourceFromMission(
+    const backend::ResourceType input_resource_type,
+    const vi_map::MissionIdList& mission_ids, const vi_map::VIMap& vi_map) {
+  CHECK(!mission_ids.empty());
+  CHECK(!FLAGS_vis_pointcloud_mission_id.empty());
+  CHECK(!FLAGS_vis_pointcloud_mission_id_topic.empty());
+
+  srand(time(NULL));
+
+  for (const vi_map::MissionId& mission_id : mission_ids) {
+    const std::string& mission_string = mission_id.shortHex();
+    if (mission_string != FLAGS_vis_pointcloud_mission_id) {
+      continue;
+    }
+    resources::PointCloud accumulated_point_cloud_G;
+    createAndAppendAccumulatedPointCloudMessageForMission(
+        input_resource_type, mission_id, vi_map, &accumulated_point_cloud_G);
+
+    sensor_msgs::PointCloud2 ros_point_cloud_G;
+    backend::convertPointCloudType(
+        accumulated_point_cloud_G, &ros_point_cloud_G);
+    publishPointCloudInGlobalFrame(
+        FLAGS_vis_pointcloud_mission_id_topic, &ros_point_cloud_G);
+
+    return;
+  }
+  VLOG(1) << "No mission found with id: " << FLAGS_vis_pointcloud_mission_id;
+}
+
+void visualizeReprojectedDepthResourceSequentially(
+    const backend::ResourceType input_resource_type,
+    const vi_map::MissionIdList& mission_ids, const vi_map::VIMap& vi_map) {
+  CHECK(!mission_ids.empty());
+  srand(time(NULL));
+
+  // Extract resource information from the missions.
+  std::map<int64_t, SequentialPointCloud> incremental_resources;
+  std::vector<std::string> mission_topic_prefix;
+  uint16_t mission_counter = 0u;
+  const std::string mission_prefix = "mission_";
+  for (const vi_map::MissionId& mission_id : mission_ids) {
+    const std::string& mission_string = mission_id.shortHex();
+    VLOG(1) << "mission string is " << mission_string;
+    if (!FLAGS_vis_pointcloud_mission_id.empty() &&
```

Comment:
I think it would make more sense to me to also here move this mission selection up to the calling function which would either pass what I assume is all mission_ids or a list with a single mission id. Otherwise the function interface is a bit confusing since depending on the flag it might ignore the input arguments.

Reply:
in fact, we don't need this flag at all! the visualization plugin already has a flag that can filter missions. -- removed this part.